File: JavaScript/6kyu/Who likes it?.js
function likes(a) {
  return a.length===1 
    ? `${a[0]} likes this` 
  : a.length===2 
    ? `${a[0]} and ${a[1]} like this`
  : a.length===3 
    ? `${a[0]}, ${a[1]} and ${a[2]} like this`
  : a.length>=4 
    ? `${a[0]}, ${a[1]} and ${a.length-2} others like this`
  : `no one likes this`
}

File: Java/6kyu/Who likes it?.java
class Solution {
    public static String whoLikesIt(String... a) {
        System.out.println(a);
        if(a.length==1) return String.format("%s likes this",a[0]);
        if(a.length==2) return String.format("%s and %s like this", a[0],a[1]);  
        if(a.length==3) return String.format("%s, %s and %s like this",a[0],a[1],a[2]);
        if(a.length>=4) return String.format("%s, %s and %s others like this",a[0],a[1],String.valueOf(a.length-2));
        return "no one likes this";
    }
}
